core/src/impl/Kokkos_SharedAlloc.cpp

```diff
@@ -285,50 +285,55 @@ void SharedAllocationRecord<void, void>::print_host_accessible_records(
   // allocation.
   const SharedAllocationRecord<void, void>* r = root->m_next;
 
-  char buffer[256];
-
-  if (detail) {
-    while (r != root) {
-      // Formatting dependent on sizeof(uintptr_t)
-      const char* format_string;
-
-      if (sizeof(uintptr_t) == sizeof(unsigned long)) {
-        format_string =
-            "%s addr( 0x%.12lx ) list( 0x%.12lx 0x%.12lx ) extent[ 0x%.12lx + "
-            "%.8ld ] count(%d) dealloc(0x%.12lx) %s\n";
-      } else if (sizeof(uintptr_t) == sizeof(unsigned long long)) {
-        format_string =
-            "%s addr( 0x%.12llx ) list( 0x%.12llx 0x%.12llx ) extent[ "
-            "0x%.12llx + %.8ld ] count(%d) dealloc(0x%.12llx) %s\n";
-      }
+  std::vector<char> buffer(256);
 
-      snprintf(buffer, 256, format_string, space_name,
-               reinterpret_cast<uintptr_t>(r),
-               reinterpret_cast<uintptr_t>(r->m_prev),
-               reinterpret_cast<uintptr_t>(r->m_next),
-               reinterpret_cast<uintptr_t>(r->m_alloc_ptr), r->m_alloc_size,
-               r->use_count(), reinterpret_cast<uintptr_t>(r->m_dealloc),
-               r->m_alloc_ptr->m_label);
-      s << buffer;
-      r = r->m_next;
-    }
-  } else {
-    while (r != root) {
-      // Formatting dependent on sizeof(uintptr_t)
-      const char* format_string;
-
-      if (sizeof(uintptr_t) == sizeof(unsigned long)) {
-        format_string = "%s [ 0x%.12lx + %ld ] %s\n";
-      } else if (sizeof(uintptr_t) == sizeof(unsigned long long)) {
-        format_string = "%s [ 0x%.12llx + %ld ] %s\n";
-      }
+  while (r != root) {
+    if (detail) {
```

Comment:
Is `detail` updated somewhere during the execution of the while loop?
Seeing that the if-else blocks are largely the same other than `format_strings` within each lambda, I wonder if this condition should be placed within lambda for ternary operator and consolidate these if-else blocks.

Reply:
`detail` is not changed. I tried to combine both code paths but ultimately it seemed best to group the format strings for `detail` and non-`detail` separately together with the `snprintf` command. Feel free to have a go if you have a better idea. We could also remove/deprecate the `detail` path or something like that but it's not clear to me how much effort it's worth spending on it at this point. Ultimately, this pull request is only for getting the `GitHub` CI running again.